Rename sanitycheck -> twister

nashif opened the pull request:
This collection of commits renames sanitycheck script and anything related to that to the new name:

# twister #

Reason for this rename is to comply with inclusive language guidelines.

For reference: https://en.wikipedia.org/wiki/Tornado

nashif committed:
sanitycheck: move schemas to scripts/schemas

Move yaml schemas into one single location. We have quite a few schema
used across the tree, one common place will make them easier to find and
reference.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: move release data to release/

Move the CSV files with release related data out of the python directory
into scripts/releases, which is more appropriate.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: remove obsolete .gitignore

Remove obsolete .gitignore file

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: move sanity_chk -> pylib/twister

Move the "module" classes used by sanitycheck into a new location under
pylib, where we can have other python libraries reside instead of having
them all over the place.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
sanitycheck: rename main script

We are now known as twister.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
twister: change all occurances of sanitycheck

mass search/replace for sanitycheck replacing it with the new name:
twiser.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
doc: rename sanitycheck -> twister

Change documentation to use new name for sanitycheck.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
CODEOWNERS: rename sanitycheck

Also replace things in CODEOWNER file.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
maintainers: rename sanitycheck

Change sanitycheck occurances with twister.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
twister: move unit tests

Move all unittests to a new folder with the twister name.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
twister: adapt platform schema

Adapt schema to use twister instead of sanitycheck.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
twister: rename in code

Some code had reference to sanitycheck, replace with twister.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
twister: rename in cmake/

Some extenstions in cmake used sanitycheck name, replace those with
twister.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
gitignore: add twister-out

Add twister-out in the main .gitignore file.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
ci: rename sanitycheck -> twister

Change all CI scripts to use the new script name: twister.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
twister: tests: rename sanitycheck -> twister

Change all tests to use twister instead of sanitycheck.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
modules: rename options for twister support

Zephyr modules with test support should also be using twister instead of
sanitycheck.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
twister: fixed pylint issues

Fixed import and unused variable warning.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

tyrantlizard77 reviewed (commented):
consider adding a deprecated "sanitycheck" wrapper script which just prints a warning "This was renamed to twister" and invokes twister with the provided args, so that users aren't taken completely by surprise when the script seems to no longer exist. and then remove the placeholder script one release later (2.6 LTS). just a thought.

jenmwms commented on .buildkite/run.sh:
```diff
@@ -6,13 +6,13 @@ set -eE
-	buildkite-agent artifact upload sanitycheck-${BUILDKITE_JOB_ID}.xml
+	mv sanity-out/twister.xml twister-${BUILDKITE_JOB_ID}.xml
+	buildkite-agent artifact upload twister-${BUILDKITE_JOB_ID}.xml
 
```
Are these remaining `sanity-out` intentional, or does it need to be renamed too?

jenmwms commented on .gitignore:
```diff
@@ -32,6 +32,7 @@ doc/samples
 doc/latex
 doc/themes/zephyr-docs-theme
 sanity-out*
```
Is this remaining `sanity-out` intentional, or does it need to be renamed too?

jenmwms commented on doc/guides/coverage.rst:
```diff
@@ -132,24 +132,24 @@ You may postprocess these with your preferred tools. For example:
-    $ sanitycheck --coverage -p native_posix -T tests/bluetooth
+    $ twister --coverage -p native_posix -T tests/bluetooth
 
 which will produce ``sanity-out/coverage/index.html`` with the report.
```
Does this need to be renamed too?

jenmwms commented on scripts/ci/run_ci.sh:
```diff
@@ -94,14 +94,14 @@ function on_complete() {
+	if [ -e ./sanity-out/twister.xml ]; then
+		echo "Copy ./sanity-out/twister.xml"
+		cp ./sanity-out/twister.xml shippable/testresults/
 	fi
```
Similar to .buildkite/run.sh - does it need to be renamed too?

yerabolu reviewed (commented):
report_name in needs to be changed as well to address all the report names.

yerabolu commented:
> report_name in needs to be changed as well to address all the report names.

Sorry, ignore this comment. changes to report names are already in.

tejlmand commented on doc/contribute/index.rst:
```diff
@@ -293,23 +293,23 @@ Note, gitlint only checks HEAD (the most recent commit), so you should run it
 
-sanitycheck
+twister
 ===========
```
Need to shorten this:
```suggestion
=======
```

nashif committed:
twister: add a wrapper script with old name

Add a simpel wrapper script called 'sanitycheck' to be removed after
Zephyr 2.5.

This script has a warning and a 5 sec delay to get the attention of the
caller.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif commented:
> consider adding a deprecated "sanitycheck" wrapper script which just prints a warning "This was renamed to twister" and invokes twister with the provided args, so that users aren't taken completely by surprise when the script seems to no longer exist. and then remove the placeholder script one release later (2.6 LTS). just a thought.

done

MaureenHelm commented on doc/contribute/index.rst:
```diff
@@ -293,23 +293,23 @@ Note, gitlint only checks HEAD (the most recent commit), so you should run it
 
 .. note::
-   sanitycheck does not currently run on Windows.
+   twiser does not currently run on Windows.
```
s/twiser/twister/

MaureenHelm reviewed (commented):
Looks good, just one nit
